### PR TITLE
Add spell check to website CI workflow

### PR DESCRIPTION
**Pull request by jorytindall**
### :pushpin: Summary

This PR adds an additional job to the website CI workflow that runs `cspell` spell check. Pushing this as a test to see if it works, there are spelling errors in this branch.

### :link: External links

Jira ticket: [HDS-1337](https://hashicorp.atlassian.net/browse/HDS-1337)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-1337]: https://hashicorp.atlassian.net/browse/HDS-1337?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Dhaulagiri** [comment]
👏  nicely done! The implementation looks good to me.

In order to merge this we'll need to:
* get the failing issues resolved 
* Document and socialize to the team how this change impacts their workflow (i.e. how to fix issues beyond simple spelling mistakes)

**Comment by jorytindall** [comment]
@Dhaulagiri I fixed the relevant spelling errors and added a section to the wiki in `wiki/Website-Editing-existing-content.md`

File: website/docs/patterns/button-alignment/partials/overview/overview.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ While these guidelines don’t cover every scenario, they are intended to aid in
     - In-page forms
     - The "window" in a general sense
 
-This differentiation is important because the expectations from the user are different in each context. A constrained component or pattern will often be used in an interstitial or circumstantial experience; i.e. an experience that is triggered by the user or the application illiciting a quick, progressive action.
+This differentiation is important because the expectations from the user are different in each context. A constrained component or pattern will often be used in an interstitial or circumstantial experience; i.e. an experience that is triggered by the user or the application and elicits a quick, progressive action.
```
Comment (MelSumner) [inline website/docs/patterns/button-alignment/partials/overview/overview.md:19]: This page should not be in this PR

File: website/docs/patterns/button-alignment/partials/specifications/order.md [hunk 2]
```diff
@@ -64,7 +64,7 @@ A possible exception to this ordering method is in multi-step and progressive fl
 
 As the viewport size contracts, buttons should stack vertically in accordance with their order in the [DOM (Document Object Model)](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Introduction). 
 
-- Naturally adheres to our ordering guidelines; button order remains the same except the container is vertical rather than horiztonal.
+- Naturally adheres to our ordering guidelines; button order remains the same except the container is vertical rather than horizontal.
```
Comment (MelSumner) [inline website/docs/patterns/button-alignment/partials/specifications/order.md:67]: This page should not be in this PR

**Review by MelSumner — CHANGES_REQUESTED** [review]
I think a couple of commits accidentally made it into this PR, those will need to be separated out please.

**Review by MelSumner — APPROVED** [review]
Assuming the page changes are fixes that the spelling tool found, gtg!

**Pull request merged**